packages/graphback-cli/src/components/generate.ts

```diff
@@ -49,15 +50,37 @@ export const generateUsingPlugins = async (cliFlags: CliFlags) => {
   };
   const debouncedExec = debounce(runGeneration, 100);
 
+  const modelPath = getModelPath(config.dirpath, graphbackConfig.model)
+
   if (cliFlags.watch) {
-    chokidar.watch(graphbackConfig.model, {
+    chokidar.watch(modelPath, {
       persistent: true,
       cwd: config.dirpath,
     }).on('all', debouncedExec);
   } else {
-    const schemaDocument = await project.loadSchema(join(config.dirpath, graphbackConfig.model, '/**/*.graphql'));
+    
+    const schemaDocument = await project.loadSchema(modelPath);
     const generator = new GraphbackGenerator(schemaDocument, graphbackConfig);
     generator.generateSourceCode();
   }
 };
 
+/**
+ * 
+ * Formats the model path or list of paths to ensure
+ *   * only paths from inside the project are allowed
+ *   * if a directory is supplied *.graphql is appended
+ * 
+ * @param baseDir the base directory that should be appended to the path or paths
+ * @param model a path or a list of paths
+ */
+function getModelPath(baseDir: string, model: string | Array<string>): string | Array<string> {
+  if (typeof model === 'string' && existsSync(model) && lstatSync(model).isDirectory()) {
+    return join(baseDir, model, '/**/*.graphql')
```

Comment:
Should we assume that

 **(a)** a basic `./model` directory means search `./model` and all of its subdirectories, or 
**(b)** should specifying `./model` mean only look in the `model` folder itself?

From a declarative point and just looking at the config without referring to the documentation it should be **(b)**.

This may break compatibility but if is wrong it's better to do this now before v1. What do you think?

I found this issue when trying out an example app on GitHub. The folder structure caused Graphback's generation to pick up `datamodel.graphql` and `generatedSchema.graphql` getting picked up as the data model because `./src` was the only model path that worked, causing some information to be lost from the final schema.

```
- src
  - datamodel.graphql
  - schema
      - generatedSchema.graphql
```

Reply:
I would argue that the particular case you describe is a misconfiguration on the user's part but at the same time, I'm totally happy to go with option **b**. While it technically breaks compatibility, None of the current templates place things in subdirectories.

Reply:
If we phrase `./model` as corner case and backwards compatibility mode we should not care much as users will specify exactly what they looking for.

Reply:
Let's go with **a** then, which is what you have :) Just docs updates are needed now.